File: JAVA/1614. Maximum Nesting Depth of the Parentheses.java
class Solution {
    public int maxDepth(String s) {
        int count=0,maxdept=0;
        for(int i=0;i<s.length();i++)
        {
            if(s.charAt(i) == '(')
            {
                count++;
            }
            else if(s.charAt(i) == ')')
            {
                count--;
            }
            maxdept = Math.max(count,maxdept);
        }
        return maxdept;
        
    }
}

File: C++/1614. Maximum Nesting Depth of the Parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int i=0;
        int temp=0,count=0;
        stack <int> st;
        int n=s.length();
        while(i<n)
        {
            if(s[i]=='(')
            {
            st.push(s[i]);
            temp++;
            }
            else if(s[i]==')')
            {
                st.pop();
                temp--;
            }
            if(count<temp)
            count=temp;
            i++;
        }
        return count;
    }
};
